### PR TITLE
Add KeywordValue

### PR DESCRIPTION
**Pull request by nattridgell**
- adds KeywordValue (a type of Style value)
- adds KeywordValue.StyleValueKeyword 'enum'
- adds KeywordValue.isKeywordValue function for checking if
  a value is a keyword.
- some basic test cases for KeywordValue

### CODE REVIEW

File: src/number-value.js [hunk 1]
```diff
@@ -12,27 +12,28 @@
 //     See the License for the specific language governing permissions and
 // limitations under the License.
 
-(function(shared, scope) {
+(function(shared, scope, testing) {
```
Comment (wilddamon) [inline src/number-value.js]: Since these changes are unrelated, it's good practice to put them in a different PR.

**Comment by wilddamon** [comment]
LGTM, one minor comment

**Comment by nattridgell** [comment]
Split into 2 PRs.  Sorry.  Won't do that again. :-)
